tequilapi/http_middlewares.go

```diff
@@ -24,37 +24,59 @@ import (
 
 type corsHandler struct {
 	originalHandler http.Handler
+	corsConfig      CorsConfig
+}
+
+// CorsConfig allows customizing CORS (Cross-Origin Resource Sharing) behaviour - whitelisting only specific domains
+type CorsConfig struct {
+	DefaultTrustedOrigin  string
+	AllowedOriginSuffixes []string
```

Comment:
A proposal for design improvement.
Instead of passing around "naked" CorsConfig struct without methods, create an interface with single method: AllowedOrigin(..requestedOrigin..), that way we can have different implementations for different cases like AllowAll, AllowSpecific, etc., corsHandler can depend on this interface only and put returned value inside headers when needed.